test/cpp/end2end/xds/xds_outlier_detection_end2end_test.cc

```diff
@@ -194,7 +202,8 @@ TEST_P(OutlierDetectionTest, SuccessRateMaxPercent) {
                           .set_server_expected_error(StatusCode::CANCELLED));
   CheckRpcSendOk(DEBUG_LOCATION, 1, rpc_options2);
   CheckRpcSendOk(DEBUG_LOCATION, 1, rpc_options3);
-  gpr_sleep_until(grpc_timeout_milliseconds_to_deadline(100));
+  gpr_sleep_until(grpc_timeout_milliseconds_to_deadline(
```

Comment:
Since the problem here is that the timing isn't lining up properly, I think we should try to remove as many of these sleeps as possible.  We should be able to replace almost all of them with `WaitForBackend()` calls to detect when the ejection actually occurs.  I think the only cases where that won't work are the ones where no ejection is happening, so those cases will probably still need sleeps, but those should also be less prone to flakiness.

Reply:
I'm not sure if we can wait for backends elsewhere easily. FYI, I did not see time flakes in 10K runs otherwise.

- These tests check for no ejection, hence sleep seems to be the only option.
SuccessRateStdevFactor
SuccessRateEnforcementPercentage
SuccessRateMinimumHosts
SuccessRateRequestVolume
FailurePercentageThreshold
FailurePercentageEnforcementPercentage                                                                                                                                       
FailurePercentageMinimumHosts
FailurePercentageRequestVolume
SuccessRateAndFailurePercentageBothDisabled
SuccessRateAndFailurePercentageEjectionPolicyDisabled                 

- These tests should wait for a subset of backends to be ejected, which is not easy to change. 
SuccessRateMaxPercent (one of {0, 1})
FailurePercentageMaxPercentage (one of {0, 1})
SuccessRateAndFailurePercentage (two of {0, 1, 2, 3})

Reply:
I guess I was not clear in saying not easy. Yes, above works in one hand, but these tests should also ensure if not more than expected # of backends switched, so I feel using a timeout can be justified for the same reason we use it in the first group of tests to ensure we have proper coverage for false positives.

I considered adding something similar to above followed by a timeout, but then found this does not help with extra complexity as long as the total max timeout value is the same.

WDYT?

Reply:
After the above code, you can still send 100 RPCs to each of the 3 remaining backends and make sure they all get to the right places.  That should ensure that no other backends were incorrectly ejected.

Reply:
I see that works since we know a round of update happened here, unlike the first group of tests. Will update.

Reply:
Done.